go/noise/laplace_noise_test.go

```diff
@@ -172,6 +172,525 @@ func TestDeltaForThresholdLaplace(t *testing.T) {
+	for _, tc := range []struct {
+		desc                     string
+		mean, lambda, prob, want float64
+	}{
+		// 2 high-precision calculated random tests.
+		{
+			desc:   "High-precision random test",
+			mean:   64,
+			lambda: 4,
+			prob:   0.7875404240919761168041,
+			want:   67.4234254367},
+		{
+			desc:   "High-precision random test",
+			mean:   23,
+			lambda: 2,
+			prob:   0.1479685611330654517049,
+			want:   20.564783456,
+		},
+		// Edge cases where p = 0 or p = 1.(Output should be infinite)
+		{
+			desc:   "Negative infinity output when probablity is 0",
+			mean:   0,
+			lambda: 1,
+			prob:   0,
+			want:   math.Inf(-1),
+		},
+		{
+			desc:   "Positive infinity output when probablity is 1",
+			mean:   1,
+			lambda: 1,
+			prob:   1,
+			want:   math.Inf(1),
+		},
+		// Logical testing (with p = 0.5, return value should be mean regardless of lambda)
+		{
+			desc:   "%50 confidence level, different lambda",
+			mean:   5,
+			lambda: 5,
+			prob:   0.5,
+			want:   5,
+		},
+		{
+			desc:   "%50 confidence level, different lambda",
+			mean:   5,
+			lambda: 10,
+			prob:   0.5,
+			want:   5,
+		},
+		// Edge cases where probablity is low or high. (Tests for accuracy)
+		{
+			desc:   "Low probablity",
+			mean:   0,
+			lambda: 3,
+			prob:   2.88887425971E-8,
+			want:   -50,
+		},
+		{
+			desc:   "High probablitiy",
+			mean:   0,
+			lambda: 3,
+			prob:   0.999999971111257,
+			want:   50,
+		},
+	} {
+		got := inverseCDFLaplace(tc.mean, tc.lambda, tc.prob)
+		if !approxEqual(got, tc.want) {
+			t.Errorf("TestInverseCDFLaplace(%f,%f,%f)=%0.12f, want %0.12f, desc: %s", tc.mean, tc.lambda,
+				tc.prob, got, tc.want, tc.desc)
+		}
+	}
+
+}
+
+func TestConfidenceIntervalLaplace(t *testing.T) {
+	// Tests for getConfidenceIntervalLaplace function
+	for _, tc := range []struct {
+		desc                    string
+		noisedValue             float64
+		lambda, confidenceLevel float64
+		want                    ConfidenceInterval
+	}{
+		// 4 Random pre-calculated tests.
+		{
+			desc:            "Random test",
+			noisedValue:     13,
+			lambda:          27.33333333333,
+			confidenceLevel: 0.95,
+			want:            ConfidenceInterval{-68.88334881, 94.88334881},
+		},
+		{
+			desc:            "Random test",
+			noisedValue:     83.1235,
+			lambda:          60,
+			confidenceLevel: 0.76,
+			want:            ConfidenceInterval{-2.503481338, 168.7504813},
+		},
+		{
+			desc:            "Random test",
+			noisedValue:     5,
+			lambda:          6.6666666666667,
+			confidenceLevel: 0.4,
+			want:            ConfidenceInterval{1.594495842, 8.405504158},
+		},
+		{
+			desc:            "Random test",
+			noisedValue:     65.4621,
+			lambda:          700,
+			confidenceLevel: 0.2,
+			want:            ConfidenceInterval{-90.73838592, 221.6625859},
+		},
+		// Near 0 and 1 confidence levels.
+		{
+			desc:            "Low confidence level",
+			noisedValue:     50,
+			lambda:          10,
+			confidenceLevel: 0.01,
```

Comment:
i think we should go more extreme in both cases, e.g., 10^-10 and 1-10^-10

Reply:
Added, the results for extremely high confidence level were accurate to 8 significant digits.